logdetective/server/database/models/merge_requests.py

```diff
@@ -0,0 +1,462 @@
+            comment = (
+                session.query(cls)
+                .join(
+                    GitlabMergeRequests, cls.merge_request_id == GitlabMergeRequests.id
+                )
+                .filter(
+                    GitlabMergeRequests.forge == forge,
+                    GitlabMergeRequests.project_id == project_id,
+                    GitlabMergeRequests.mr_iid == mr_iid,
+                    GitlabMergeRequests.job_id == job_id,
+                    cls.comment_id == comment_id,
+                )
+                .first()
+            )
+
+            return comment
+
+    @classmethod
+    def get_latest_comment(
+        cls,
+        forge: Forge,
+        project_id: int,
+        mr_iid: int,
+        job_id: int,
+    ) -> Optional["Comments"]:
+        """Search for the latest comment in the merge request.
+
+        Args:
+          forge: forge name
+          project_id: forge project id
+          mr_iid: merge request forge iid
+          job_id: forge job id
+        """
+        with transaction(commit=False) as session:
+            comment = (
+                session.query(cls)
+                .join(
+                    GitlabMergeRequests, cls.merge_request_id == GitlabMergeRequests.id
+                )
+                .filter(
+                    GitlabMergeRequests.forge == forge,
+                    GitlabMergeRequests.project_id == project_id,
+                    GitlabMergeRequests.mr_iid == mr_iid,
+                    GitlabMergeRequests.job_id == job_id,
+                )
+                .order_by(desc(cls.created_at))
+                .first()
+            )
+
+            return comment
+
+    @classmethod
+    def get_or_create(  # pylint: disable=too-many-arguments disable=too-many-positional-arguments
+        cls,
+        forge: Forge,
+        project_id: int,
+        mr_iid: int,
+        job_id: int,
+        comment_id: int,
+    ) -> Optional["Comments"]:
+        """Search for a detailed comment
+        or create a new one if not found.
+
+        Args:
+          forge: forge name
+          project_id: forge project id
+          mr_iid: merge request forge iid
+          job_id: forge job id
+          comment_id: forge comment id
+        """
+        comment = Comments.get_by_details(forge, project_id, mr_iid, job_id, comment_id)
+        if comment is None:
+            id_ = Comments.create(forge, project_id, mr_iid, job_id, comment_id)
+            comment = GitlabMergeRequests.get_by_id(id_)
+        return comment
+
+
+class Reactions(Base):
+    """Store and count reactions received for
+    logdetective comments"""
+
+    __tablename__ = "reactions"
+
+    id = Column(Integer, primary_key=True)
+    comment_id = Column(
+        Integer,
+        ForeignKey("comments.id"),
+        nullable=False,
+        index=True,
+        comment="The associated comment id",
+    )
+    reaction_type = Column(
+        String(50),  # e.g. 'thumbs-up'
```

Comment:
I went down the rabbit-hole on this, to discover that the current longest emoji supported by Gitlab is `:couple_with_heart_person_person_medium_skin_tone_dark_skin_tone:` which is 66 characters long. We should probably make this string a bit longer. Maybe 100 to be on the safe side?

Reply:
```suggestion
        String(127),  # e.g. 'thumbs-up'
```

127 or 255 perhaps?